src/frontend/src/optimizer/plan_node/expr_rewritable.rs

```diff
@@ -18,31 +18,40 @@ use paste::paste;
 
 use super::*;
 use crate::expr::ExprRewriter;
-use crate::{for_batch_plan_nodes, for_stream_plan_nodes};
+use crate::for_stream_plan_nodes;
 
 /// Rewrites expressions in a `PlanRef`. Due to `Share` operator,
 /// the `ExprRewriter` needs to be idempotent i.e. applying it more than once
 /// to the same `ExprImpl` will be a noop on subsequent applications.
 pub trait ExprRewritable {
-    fn rewrite_exprs(&self, r: &mut dyn ExprRewriter) -> PlanRef;
+    fn has_rewritable_expr(&self) -> bool {
+        false
+    }
```

Comment:
Why is this needed? (compared with a no-op `rewrite_exprs` impl)

Reply:
It's just so we don't have to clone the plan.